daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs

```diff
@@ -216,3 +236,79 @@ runLedgerNavigator flags remainingArguments = do
           ]
         , ["  }"]
         ]
+
+----------------
+-- HTTP JSON API
+----------------
+
+newtype Method = Method
+  { unMethod :: BS.ByteString
+  } deriving IsString
+
+newtype Path = Path
+  { unPath :: BS.ByteString
+  } deriving IsString
+
+-- | Run a request against the HTTP JSON API.
+httpJsonRequest :: LedgerArgs -> Method -> Path -> IO (Response A.Value)
+httpJsonRequest LedgerArgs {..} method path = do
+  resp <-
+    httpJSON $
+    setRequestPort port $
+    setRequestHost (BSC.pack host) $
+    setRequestMethod (unMethod method) $
+    setRequestPath (unPath path) $
+    setRequestHeader
+      "authorization"
+      [BSC.pack $ sanitizeToken tok | Just (L.Token tok) <- [tokM]]
+      defaultRequest
+  let status = getResponseStatusCode resp
+  unless (status == 200) $
+    fail $ "Request failed with error code " <> show status
+  pure resp
+
+-- This matches how the com.daml.ledger.api.auth.client.LedgerCallCredentials
+-- behaves.
+sanitizeToken :: String -> String
+sanitizeToken tok
+  | "Bearer " `isPrefixOf` tok = tok
+  | otherwise = "Bearer " <> tok
+
+listPartiesJson :: LedgerArgs -> IO [PartyDetails]
+listPartiesJson args = do
+  body <- getResponseBody <$> httpJsonRequest args "GET" "/v1/parties"
+  either (fail . show) return $ do
+    ps <- getObject A._Array "result" body
+    traverse getPartyDetails $ Vector.toList ps
+
+getPartyDetails :: A.Value -> Perhaps PartyDetails
```

Comment:
Why not define a `FromJSON` instance for `PartyDetails` or a newtype thereof? The `lens-aeson` stuff doesn’t seem all that benefitial here.

Reply:
I think that would be way nicer. But that would result in an orphan instance, unless I write the instance into the hs bindings. What's your preference here?

Reply:
I don’t particularly mind orphan instances in applications. They’re bad in libraries but we don’t distribute our code. You could also newtype it. I don’t have anything against putting it in the HS bindings either. Whatever is easiest.

Reply:
Ok, I added the to hs bindings.